sdk/batch/azure-compute-batch/src/main/java/com/azure/compute/batch/implementation/task/TaskManager.java

```diff
@@ -69,41 +73,45 @@ public Exception getException() {
          *
          * @param taskList The list of tasks to be submitted.
          */
-        private void submitChunk(List<BatchTaskCreateContent> taskList) {
+        private void submitChunk(List<BatchTaskCreateParameters> taskList) {
             try {
-                taskSubmitter.submitTasks(jobId, new BatchTaskGroup(taskList)).doOnError(e -> {
-                    if (e instanceof HttpResponseException) {
-                        // Handle HttpResponseException
-                        handleException((HttpResponseException) e, taskList);
-                    } else {
-                        // Handle generic exceptions
-                        exception = (Exception) e;
-                        pendingList.addAll(taskList);
-                    }
-                }).subscribe(response -> {
-                    if (response != null && response.getValue() != null) {
-                        for (BatchTaskAddResult result : response.getValue()) {
-                            if (result.getError() != null) {
-                                if (result.getStatus() == BatchTaskAddStatus.SERVER_ERROR) {
-                                    // Server error will be retried
-                                    for (BatchTaskCreateContent batchTaskToCreate : taskList) {
-                                        if (batchTaskToCreate.getId().equals(result.getTaskId())) {
-                                            pendingList.add(batchTaskToCreate);
-                                            break;
-                                        }
+                BatchCreateTaskCollectionResult response
+                    = taskSubmitter.submitTasks(jobId, new BatchTaskGroup(taskList));
+
+                if (response != null && response.getValues() != null) {
+                    for (BatchTaskCreateResult result : response.getValues()) {
+                        if (result.getError() == null) {
+                            continue; // success
+                        }
+
+                        if (result.getStatus() == BatchTaskAddStatus.SERVER_ERROR) {
+                            // Requeue this single task for retry
```

Comment:
This comment seems off. This doesn't requeue a single task, it requeues every task referenced in the response body.